great_expectations/datasource/data_connector/util.py

```diff
@@ -76,39 +76,40 @@ def batch_definition_matches_batch_request(
 def map_data_reference_string_to_batch_definition_list_using_regex(
     datasource_name: str,
     data_connector_name: str,
-    data_asset_name: str,
     data_reference: str,
     regex_pattern: str,
     group_names: List[str],
+    data_asset_name: Optional[str] = None,
 ) -> Optional[List[BatchDefinition]]:
-    batch_request: BatchRequest = (
-        convert_data_reference_string_to_batch_request_using_regex(
-            data_reference=data_reference,
-            regex_pattern=regex_pattern,
-            group_names=group_names,
-        )
+    processed_data_reference: Optional[
+        Tuple[str, PartitionDefinitionSubset]
+    ] = convert_data_reference_string_to_partition_definition_using_regex(
+        data_reference=data_reference,
+        regex_pattern=regex_pattern,
+        group_names=group_names,
     )
-    if batch_request is None:
+    if processed_data_reference is None:
         return None
-
+    data_asset_name_from_partition_definition: str = processed_data_reference[0]
+    partition_definition: PartitionDefinitionSubset = processed_data_reference[1]
     if data_asset_name is None:
-        data_asset_name = batch_request.data_asset_name
+        data_asset_name = data_asset_name_from_partition_definition
 
     return [
         BatchDefinition(
             datasource_name=datasource_name,
             data_connector_name=data_connector_name,
             data_asset_name=data_asset_name,
-            partition_definition=PartitionDefinition(batch_request.partition_request),
+            partition_definition=PartitionDefinition(partition_definition),
         )
     ]
 
```

Comment:
i know this may be super minor, but this part where we directly convert `partition_request` to `PartitionDefinition` always bugged me a bit. Thanks for this change :)